course_catalog/etl/youtube.py

```diff
@@ -0,0 +1,105 @@
+"""video catalog ETL"""
+import logging
+from django.conf import settings
+from googleapiclient.discovery import build
+import googleapiclient.errors
+import yaml
+import requests
+
+
+YOUTUBE_API_SERVICE_NAME = "youtube"
+YOUTUBE_API_VERSION = "v3"
+
+log = logging.getLogger()
+
+
+def get_youtube_client():
+    """
+    Function to generate a Google api client
+
+    Returns:
+        Google Api client object
+    """
+
+    developer_key = settings.YT_DEVELOPER_KEY
+    return build(
+        YOUTUBE_API_SERVICE_NAME, YOUTUBE_API_VERSION, developerKey=developer_key
+    )
+
+
+def get_videos_for_channel(youtube_client, channel_id, offered_by):
+    """
+    Function which returns a generator that loops through a youtube channel and
+    returns video data
+
+    Args:
+        youtube_client: Youtube api client
+        channel_id: Youtube's id for a channel
+        offered_by: Our offered by tag for the youtube channel
+
+    Returns:
+        A generator that yields tuples with offered_by and video data
+
+    """
+
+    page_token = ""
+
+    while True:
+
+        try:
```

Comment:
I update the code to log and error but not break if a request for a specific channel fails, likely because of a bad channel id